### PR TITLE
fixes #48 added support for multiple MySQL flavors

### LINKED ISSUE #48: Multi Source Replication check is not working for multi source in MySQL 5.7
Hi,
In continuing to the conversation in the link below, 
This is a highly desired feature and I would like to know if you have an estimation when it will be pushed.
https://github.com/sensu-plugins/sensu-plugins-mysql/commit/68aef5b5ee20c4d880deafd5aeb7c2a0cad9099c#diff-ebbf24d20eee63e56e38316772592ed5R109

Thanks,
Eyal

### PR DESCRIPTION
**Pull request by radarnex**
#48 

#### General

- [x] Update Changelog following the conventions laid out on [Keep A Changelog](http://keepachangelog.com/)

- [ ] Update README with any necessary configuration snippets

- [ ] Binstubs are created if needed

- [ ] RuboCop passes

- [ ] Existing tests pass 

#### Purpose
Add support for multi-source replication on multiple MySQL flavors. Sorry for my ruby :-(, any comment is welcome.

#### Known Compatablity Issues
None

### CODE REVIEW

**Comment by majormoses** [comment]
@radarnex thanks I will take a closer look later, can you fix ci though?

File: bin/check-mysql-replication-status.rb [hunk 1]
```diff
@@ -103,10 +103,26 @@ def run
     begin
       db = Mysql.new(db_host, db_user, db_pass, nil, config[:port], config[:socket])
 
+      db_flavor = ''
```
Comment (majormoses) [inline bin/check-mysql-replication-status.rb]: this logic feels like it should all be its own function.

File: bin/check-mysql-replication-status.rb [hunk 2]
```diff
@@ -103,10 +103,26 @@ def run
...
+          db_flavor = 'unknown'
+        end
+      end
+
       results = if db_conn.nil?
                   db.query 'SHOW SLAVE STATUS'
-                else
+                elsif db_flavor.include? 'mariadb'
                   db.query "SHOW SLAVE '#{db_conn}' STATUS"
+                elsif db_flavor.include? 'mysql' || 'percona'
```
Comment (majormoses) [inline bin/check-mysql-replication-status.rb]: This does not do what you expect it to:
```ruby
irb(main):002:0> 'mysql'.include? 'percona' || 'mysql'
=> false
```

What you want is this:
```ruby
elsif db_flavor.include? 'mysql' || db_flavir.include? 'percona'
```

**Comment by majormoses** [comment]
@radarnex planning on picking this back up?

**Comment by radarnex** [comment]
Hi @majormoses. Lately I've been kind busy, i will try to have it this week

**Pull request closed without merging**